File: tests/expectations/tag/enum_self.c
#include <stdarg.h>
#include <stdbool.h>
#include <stdint.h>
#include <stdlib.h>

struct Foo_Bar {
  const int32_t *something;
};

enum Bar_Tag {
  Min,
  Max,
  Other,
};
typedef uint8_t Bar_Tag;

struct Min_Body {
  Bar_Tag tag;
  struct Foo_Bar _0;
};

struct Max_Body {
  Bar_Tag tag;
  struct Foo_Bar _0;
};

union Bar {
  Bar_Tag tag;
  struct Min_Body min;
  struct Max_Body max;
};

void root(union Bar b);

File: tests/expectations/enum_self.zig
const std = @import("std");

pub const Foo_Bar = extern struct {
   something: ?*i32,
};

pub const Bar_Tag = enum {
  Min,
  Max,
  Other,
};

pub const Bar = extern union {
  tag: Bar_Tag,
  min_tag: Bar_Tag,
  min: Foo_Bar,
  max_tag: Bar_Tag,
  max: Foo_Bar,
};

pub extern fn root(b: Bar) anyopaque;
